operator/pkg/reconciliation/construct_podtemplatespec.go

```diff
@@ -0,0 +1,481 @@
+// Copyright DataStax, Inc.
+// Please see the included license file for details.
+
+package reconciliation
+
+// This file defines constructors for k8s objects
+
+import (
+	"fmt"
+	"reflect"
+
+	api "github.com/datastax/cass-operator/operator/pkg/apis/cassandra/v1beta1"
+	"github.com/datastax/cass-operator/operator/pkg/images"
+	"github.com/datastax/cass-operator/operator/pkg/oplabels"
+	"github.com/datastax/cass-operator/operator/pkg/utils"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+const (
+	ServerConfigContainerName = "server-config-init"
+	CassandraContainerName    = "cassandra"
+	PvcName                   = "server-data"
+	SystemLoggerContainerName = "server-system-logger"
+)
+
+// calculateNodeAffinity provides a way to pin all pods within a statefulset to the same zone
+func calculateNodeAffinity(zone string) *corev1.NodeAffinity {
+	if zone == "" {
+		return nil
+	}
+	return &corev1.NodeAffinity{
+		RequiredDuringSchedulingIgnoredDuringExecution: &corev1.NodeSelector{
+			NodeSelectorTerms: []corev1.NodeSelectorTerm{
+				{
+					MatchExpressions: []corev1.NodeSelectorRequirement{
+						{
+							Key:      "failure-domain.beta.kubernetes.io/zone",
+							Operator: corev1.NodeSelectorOpIn,
+							Values:   []string{zone},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+// calculatePodAntiAffinity provides a way to keep the db pods of a statefulset away from other db pods
+func calculatePodAntiAffinity(allowMultipleNodesPerWorker bool) *corev1.PodAntiAffinity {
+	if allowMultipleNodesPerWorker {
```

Comment:
maybe we should hoist this arg and the if out of here and to the call site? would probably be more clear to read, right?

Reply:
This is pre-existing code that I didn't change.  Can we update it in a new PR?